astropy/coordinates/polarization.py

```diff
@@ -0,0 +1,258 @@
+    replace
+        Replace all mappings with this one.
+    """
+    global STOKES_VALUE_SYMBOL_MAP
+
+    original_mapping = STOKES_VALUE_SYMBOL_MAP.copy()
+    if not replace:
+        STOKES_VALUE_SYMBOL_MAP = {**original_mapping, **mapping}
+    else:
+        STOKES_VALUE_SYMBOL_MAP = mapping
+
+    yield
+
+    STOKES_VALUE_SYMBOL_MAP = original_mapping
+
+
+class StokesCoordInfo(MixinInfo):
+    # The attributes containing actual information.
+    _represent_as_dict_attrs = {"value"}
+    # Since there is only one attribute, use a column with the name to represent it
+    # (rather than as name.value)
+    _represent_as_dict_primary_data = "value"
+    # Attributes that should be presented as positional arguments to
+    # the class initializer (which takes "stokes" as an argument, not "value").
+    _construct_from_dict_args = ("value",)
+
+    @property
+    def unit(self):
+        return None
+
+    @property
+    def dtype(self):
+        return self._parent._data.dtype
+
+    @staticmethod
+    def default_format(val):
+        return f"{val.symbol}"
+
+    def new_like(self, cols, length, metadata_conflicts="warn", name=None):
+        """
+        Return a new StokesCoord instance which is consistent with the
+        input ``cols`` and has ``length`` rows.
+
+        This is intended for creating an empty column object whose elements can
+        be set in-place for table operations like join or vstack.
+
+        Parameters
+        ----------
+        cols : list
+            List of input columns
+        length : int
+            Length of the output column object
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts
+        name : str
+            Output column name
+
+        Returns
+        -------
+        col : `~astropy.coordinates.StokesCoord` (or subclass)
+            Empty instance of this class consistent with ``cols``
+
+        """
+        # Get merged info attributes like shape, dtype, format, description, etc.
+        attrs = self.merge_cols_attributes(
+            cols, metadata_conflicts, name, ("meta", "format", "description")
+        )
+
+        # Make an empty StokesCoord.
+        shape = (length,) + attrs.pop("shape")
+        data = np.zeros(shape=shape, dtype=attrs.pop("dtype"))
+        # Get arguments needed to reconstruct class
+        out = self._construct_from_dict(dict(value=data))
+
+        # Set remaining info attributes
+        for attr, value in attrs.items():
+            setattr(out.info, attr, value)
+
+        return out
+
+    def get_sortable_arrays(self):
+        """
+        Return a list of arrays which can be lexically sorted to represent
+        the order of the parent column.
+
+        For StokesCoord this is just the underlying values.
+
+        Returns
+        -------
+        arrays : list of ndarray
+        """
+        return [self._parent._data]
+
+
+class StokesCoord(ShapedLikeNDArray):
```

Comment:
Personally, I'm partial to data classes — they have added introspection abilities and nice convenience functions and in py3.10 get some extra bonus abilities with descriptor attributes that will allow us to refactor the `__init__` method.

```suggestion
@dataclass(frozen=True)
class StokesCoord(ShapedLikeNDArray):
```

Reply:
At this point I would rather not. This PR is already holding up the release.